maintainers/scripts/nix-diff.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+SCRIPT_PATH=$(realpath "${BASH_SOURCE[0]}")
+[ $? != 0 ] && exit 1
+LIB="$(dirname "$SCRIPT_PATH")/lib_bash"
+[ -n "$DEBUG" ] && echo "LIB=$LIB" >&2
+. "$LIB/temp" || exit 1
+
+usage() {
+    cat <<EOF
+usage: $0 from_generation to_generation
+EOF
+    exit 1
+}
+
+[ $# -ne 2 ] && usage
+
+profile=$(readlink ~/.nix-profile)
+
+dirA="${profile}-$1-link"
+dirB="${profile}-$2-link"
+
+no_generation() {
+    local gen="$1"
+    echo no generation \`"$gen"\` found >&2
+    exit 1
+}
+
+if [[ ! -L "$dirA" ]]; then
+    no_generation "$dirA"
+elif [[ ! -L "$dirB" ]]; then
+    no_generation "$dirB"
+fi
+
+temp_make
+eval versA=$(temp_name)
+[ -n "$DEBUG" ] && echo DEBUG: "versA=$versA" >&2
+
+nix-store -q --references "$dirA" | sort > "$versA"
+
+temp_make
+eval versB=$(temp_name)
+[ -n "$DEBUG" ] && echo DEBUG: "versB=$versB" >&2
+nix-store -q --references "$dirB" | sort > "$versB"
+
+in=$(comm -3 -1 "$versA" "$versB")
+temp_make
+eval add=$(temp_name)
+[ -n "$DEBUG" ] && echo DEBUG: "add=$add" >&2
+sed -r 's:^[^-]*-(.*)$:\1+:' <(echo "$in") \
+    | grep -v env-manifest.nix \
```

Comment:
This only works for imperative user profiles (nix-env), perhaps the script name should reflect that?